cmd/internal/find/primary/kind.go

```diff
@@ -0,0 +1,70 @@
+package primary
+
+import (
+	"fmt"
+
+	"github.com/gobwas/glob"
+	"github.com/puppetlabs/wash/cmd/internal/find/parser/predicate"
+	"github.com/puppetlabs/wash/cmd/internal/find/types"
+)
+
+// Kind is the kind primary
+//
+// kindPrimary => -kind ShellGlob
+//nolint
+var Kind = Parser.add(&Primary{
+	Description:         "Returns true if the entry's kind matches the provided glob",
+	DetailedDescription: kindDetailedDescription,
+	name:                "kind",
+	args:                "glob",
+	shortName:           "k",
+	parseFunc: func(tokens []string) (types.EntryPredicate, []string, error) {
+		if len(tokens) == 0 {
+			return nil, nil, fmt.Errorf("requires additional arguments")
+		}
+		g, err := glob.Compile(tokens[0])
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid glob: %v", err)
+		}
+		return kindP(g, false), tokens[1:], nil
+	},
+})
+
+func kindP(g glob.Glob, negated bool) types.EntryPredicate {
+	p := kindPredicate{
+		EntryPredicate: types.ToEntryP(func(e types.Entry) bool {
+			// kind is a schema predicate, so the entry predicate should
+			// always return true
+			return true
+		}),
+		g: g,
+	}
+	p.SetSchemaP(types.ToEntrySchemaP(func(s *types.EntrySchema) bool {
+		if !g.Match(s.Path()) {
+			return negated
+		}
+		// g matches s.Path()
+		if negated {
+			return false
+		}
+		return true
+	}))
+	p.RequireSchema()
+	return p
+}
+
+// The separate type's necessary to implement proper Negation semantics.
+type kindPredicate struct {
+	types.EntryPredicate
+	g       glob.Glob
+	negated bool
+}
+
+func (p kindPredicate) Negate() predicate.Predicate {
+	return kindP(p.g, !p.negated)
+}
+
+// TODO: Fill this description out
+const kindDetailedDescription = `
+(-k|-kind) glob
```

Comment:
I'll flesh this out once https://github.com/puppetlabs/wash/pull/449#discussion_r313008477 is addressed.